packages/data-provider/src/controllers/dictionaryController.ts

```diff
@@ -31,9 +32,22 @@ const controller = (dependencies: BaseDependencies) => {
 					throw new BadRequest('Request is missing `version` parameter.');
 				}
 
-				const registered = await dictionaryService.register(categoryName, dictionaryName, dictionaryVersion);
+				const { dictionary, category } = await dictionaryService.register(
+					categoryName,
+					dictionaryName,
+					dictionaryVersion,
+				);
+
 				logger.info(LOG_MODULE, `Register Dictionary completed!`);
-				return res.send(registered);
+
+				const result: RegisterDictionaryResult = {
+					categoryId: category.id,
+					categoryName: category.name,
+					dictionary: dictionary.dictionary,
+					name: dictionary.name,
+					version: dictionary.version,
+				};
+				return res.send(result);
```

Comment:
Returning a response type `RegisterDictionaryResult` instead of any response from the service.